folding_aws/folding_asg_stack.py

```diff
@@ -30,3 +39,27 @@ def __init__(self, scope: core.Construct, id: str, region: str, vpc: str, ec2_in
                                 )
 
         self.asg.add_security_group(sg_ssh_in)
+
+    def get_all_azs(region: str, client: botocore.client.EC2):
+      all_azs_info = client.describe_availability_zones(Filters=[{'Name': 'region-name', 'Values': [region]}])
+      all_azs = [az['ZoneName'] for az in all_azs_info['AvailabilityZones']]
+      if len(all_azs) == 0:
+        print("Could not find any Availability Zone in region {}. \
+        Please check AWS configuration and whether this region is enabled for you".format(region))
+        sys.exit(1)
```

Comment:
I initially wrote this up in `app.py` and tested with it, but changed my mind at the last moment because that seemed like the better place for this. Copy/pasta error :(